src/sentry/static/sentry/app/utils/discover/eventView.tsx

```diff
@@ -1081,6 +1092,25 @@ class EventView {
     // we just use the default display mode
     return DisplayModes.DEFAULT;
   }
+
+  addOverrideCondition(tag: string, tagValues: string[]) {
+    if (!this.overrideConditions) {
+      this.overrideConditions = new QueryResults([]);
+    }
+    this.overrideConditions.setTagValues(tag, tagValues);
+  }
+
+  getQueryWithOverrides() {
+    const {query} = this;
+    const conditions = tokenizeSearch(query);
+    if (!this.overrideConditions) {
+      return query;
+    }
+    Object.entries(this.overrideConditions.tagValues).forEach(([tag, tagValues]) => {
+      conditions.setTagValues(tag, tagValues);
+    });
```

Comment:
Shouldn't `query` be added to the `overrideConditions`? In this case, say you have the `overrideCondition` set to `transaction.duration:<15m` and the user adds `transaction.duration<4s`, it'll be overwritten which I don't believe is the desired behaviour.

So `overrideConditions` here really should be `defaultConditions` that we add on to. Does that make sense? Or is there an argument that it should be kept the way it is presently?

Reply:
The duration case is a good point, we need to account for that. I was thinking of having it as override conditions because of parts like `event.type:transaction` which we wouldn't want to let people change.

Reply:
Nvm, leaving them as additional works fine I think.